### PR TITLE
enables to specify volumes for -mult option and fix issue in diff method

### PR DESCRIPTION
**Pull request by jcohenadad**
fixes https://github.com/neuropoly/spinalcordtoolbox/issues/1198

### CODE REVIEW

File: scripts/sct_compute_snr.py [hunk 1]
```diff
@@ -144,17 +144,29 @@ def main():
...
     elif method == 'diff':
+        # if user did not select two volumes, then exit with error
+        if not len(index_vol) == 2:
+            sct.printv('ERROR: '+str(len(index_vol))+' volumes were specified. Method "diff" should be used with exactly two volumes.', 1, 'error')
         data_1 = input_data[:, :, :, index_vol[0]]
         data_2 = input_data[:, :, :, index_vol[1]]
+        # compute voxel-average of voxelwise sum
         signal = np.mean(np.add(data_1[indexes_roi], data_2[indexes_roi]))
-        noise = np.std(np.subtract(data_1[indexes_roi], data_2[indexes_roi]))
+        # compute voxel-average of voxelwise substraction, multiplied by sqrt(2) as described in equation 7 of Dietrich et al.
```
Comment (slevyrosetti) [inline scripts/sct_compute_snr.py]: typo in the comment:
# compute STD across voxels of voxelwise substraction,...

File: scripts/sct_compute_snr.py [hunk 2]
```diff
@@ -144,17 +144,29 @@ def main():
     input_data = input_data[:, :, slices_of_interest_list, :]
     mask_data = mask_data[:, :, slices_of_interest_list]
 
+    # if user selected all slices (-vol 1), then assign index_vol
```
Comment (slevyrosetti) [inline scripts/sct_compute_snr.py]: typo in comment:
"-vol -1"

File: scripts/sct_compute_snr.py (hunk 1 above)
Comment (jcohenadad) [inline scripts/sct_compute_snr.py]: yup!

File: scripts/sct_compute_snr.py (hunk 2 above)
Comment (jcohenadad) [inline scripts/sct_compute_snr.py]: yup!

**Pull request merged**